### PR TITLE
Change user model

### PR DESCRIPTION
**Pull request by DavidHeymann**
Change user model, user interface.
added validations and tests.
change controllers and endpoints that relevant.
adds environment to groovy file and remove .env file from git.
remove all documents before each tests and drop DB before the tests run.

### CODE REVIEW

File: src/user/user.model.ts [hunk 1]
```diff
@@ -90,13 +152,14 @@ UserSchema.virtual('fullName').get(function () {
...
+/* UserSchema.pre('save', function (next) {
+  if (!this.updatedAt) this.updatedAt = new Date();
+  if (!this.createdAt) this.createdAt = new Date();
   next();
 });
 
 UserSchema.pre('update', function () {
   this.update({}, { $set: { updatedAt: new Date() } });
-});
+}); */
```
Comment (AllKinds) [inline src/user/user.model.ts]: Unnecessary comments

File: src/helpers/spec.helper.ts [hunk 2]
```diff
@@ -33,12 +33,27 @@ export const expectError = async (func: Function, params: any[]) => {
...
-  mongoose.connect(process.env.MONGODB_TEST_URI);
+  await mongoose.connect(process.env.MONGODB_TEST_URI, { useMongoClient: true });
+  const modelNames: string[] = mongoose.modelNames();
+  await cleanDatabase(modelNames);
 });
 
 beforeEach(async () => {
-  await mongoose.connection.dropDatabase();
+  const modelNames: string[] = mongoose.modelNames();
+  // await cleanDatabase(modelNames);
```
Comment (AllKinds) [inline src/helpers/spec.helper.ts]: Unnecessary comment

File: src/user/user.interface.ts [hunk 3]
```diff
@@ -1,31 +1,43 @@
...
+  directGroup: IOrganizationGroup | ObjectId;
+  managedGroup: IOrganizationGroup | ObjectId;
+  rank: Rank;
+  updatedAt: Date;
+  createdAt: Date;
 // Weak groups props
-  weakGroups: string[];
-  adminGroups: string[];
+  // weakGroups: string[];
+  // adminGroups: string[];
```
Comment (AllKinds) [inline src/user/user.interface.ts]: Unnecessary comments

File: src/user/user.spec.ts [hunk 4]
```diff
@@ -55,144 +84,313 @@ describe('Users', () => {
...
       user.should.have.property('firstName', 'Yonatan');
       user.should.have.property('lastName', 'Tal');
+      user.should.have.property('dischargeDay', userExamples[4].dischargeDay);
+      user.should.have.property('hierarchy');
+      user.hierarchy.should.have.ordered.members(['www', 'microsoft', 'github']);
+      user.should.have.property('job', 'Programmer');
       user.should.have.property('rank', 'Newbie');
-      user.should.have.property('isSecurityOfficer', false);
-      user.should.have.property('clearance', 0);
+      user.should.have.property('responsibility', 'None');
```
Comment (eladex) [inline src/user/user.spec.ts]: use enum

File: src/user/user.interface.ts [hunk 5]
```diff
@@ -1,31 +1,40 @@
 import { Document } from 'mongoose';
-import { Rank } from '../utils';
+import { Rank, Responsibility } from '../utils';
 import { IOrganizationGroup } from '../group/organizationGroup/organizationGroup.interface';
+import { ObjectId } from 'bson';
 
 export interface IUser extends Document {
 // User's Basic information
   _id: string;
```
Comment (eladex) [inline src/user/user.interface.ts]: need to be of type objectId

**Pull request merged**